src/example.c

```diff
@@ -330,7 +330,10 @@ int main(int argc, char ** argv) {
 		}
 		if (mpd_connection_get_error(conn) != MPD_ERROR_SUCCESS)
 			handle_error(conn);
-
+	} else if (argc == 3 && strcmp(argv[1], "partition") == 0) {
+		if (!mpd_run_partition(conn, argv[2]))
+			return handle_error(conn);
+		printf("switched to partition %s\n", argv[2]);
```

Comment:
creating, listing and idling on a partition i've tested successfully (using the example).
The actual switching to the new partition (with a new queue, outputs) i was unsure how to test.